server/server.go

```diff
@@ -86,11 +89,112 @@ type rodeServer struct {
+	if err != nil {
+		log.Error("failed to mget documents", zap.Error(err))
+		return err
+	}
+	if response.IsError() {
+		return fmt.Errorf("unexpected status code from mget request: %d", response.StatusCode)
+	}
+	mGetResponse := esMultiGetResponse{}
+	if err := decodeResponse(response.Body, &mGetResponse); err != nil {
+		return err
+	}
+
+	var body bytes.Buffer
+	for i := range resourceNames {
+		resourceName := resourceNames[i]
+		existingDocument := mGetResponse.Docs[i]
+		if existingDocument.Found {
+			log.Debug("skipping resource creation because it already exists", zap.String("resourceName", resourceName))
+			continue
+		}
+
+		log.Debug("Adding resource to bulk request", zap.String("resourceName", resourceName))
+
+		metadata, _ := json.Marshal(esBulkQueryFragment{
+			Create: &esBulkQueryCreateFragment{
+				Id: resourceName,
+			},
+		})
+		metadata = append(metadata, "\n"...)
+
+		data, _ := protojson.MarshalOptions{EmitUnpopulated: true}.Marshal(proto.MessageV2(&pb.GenericResource{Name: resourceName}))
+		data = append(data, "\n"...)
+
+		body.Grow(len(metadata) + len(data))
+		body.Write(metadata)
+		body.Write(data)
+	}
+
+	// no new generic resources to create
+	if body.Len() == 0 {
+		return nil
+	}
+
+	response, err = r.esClient.Bulk(
+		bytes.NewReader(body.Bytes()),
+		r.esClient.Bulk.WithContext(ctx),
+		r.esClient.Bulk.WithRefresh(r.elasticsearchConfig.Refresh.String()),
+		r.esClient.Bulk.WithIndex(rodeElasticsearchGenericResourcesIndex))
+
+	if err != nil {
+		log.Error("failed to create generic resources", zap.Error(err))
+		return fmt.Errorf("failed to create generic resources: %s", err)
+	}
+
+	if response.IsError() {
+		return fmt.Errorf("unexpected status code while creating generic resources: %d", response.StatusCode)
+	}
+
+	bulkResponse := &esBulkResponse{}
+	err = esutil.DecodeResponse(response.Body, bulkResponse)
+	if err != nil {
+		return err
+	}
+
+	var errors []error
+	for i := range bulkResponse.Items {
+		item := bulkResponse.Items[i].Create
+		if item.Error != nil && item.Status != http.StatusConflict {
+			itemError := fmt.Errorf("error creating generic resource [%d] %s: %s", item.Status, item.Error.Type, item.Error.Reason)
+			errors = append(errors, itemError)
+		}
+	}
+
+	if len(errors) > 0 {
+		log.Error("Failed to create all resources", zap.Any("errors", errors))
+		return fmt.Errorf("failed to create all resources: %v", errors)
+	}
+
+	return nil
+}
+
 func (r *rodeServer) BatchCreateOccurrences(ctx context.Context, occurrenceRequest *pb.BatchCreateOccurrencesRequest) (*pb.BatchCreateOccurrencesResponse, error) {
 	log := r.logger.Named("BatchCreateOccurrences")
 	log.Debug("received request", zap.Any("BatchCreateOccurrencesRequest", occurrenceRequest))
 
-	//Forward to grafeas to create occurrence
+	if err := r.batchCreateGenericResources(ctx, occurrenceRequest); err != nil {
```

Comment:
wonder if we're going to need to do some sort of migration to backfill existing resources from occurrences that are already in ES.  maybe we don't care, but presumably someone might already have this data in there and wouldn't also have the corresponding resources.